libvast/vast/command.hpp

```diff
@@ -148,25 +148,54 @@ class command {
 
   /// Adds a new subcommand.
   /// @returns a pointer to the new subcommand.
-  command* add(fun child_run, std::string_view child_name,
-               std::string_view child_description,
-               caf::config_option_set child_options = {});
+  command*
+  add(std::string_view child_name, caf::config_option_set child_options = {});
 
   /// Adds a new subcommand.
   /// @returns a pointer to the new subcommand.
-  command* add(fun child_run, std::string_view child_name,
-               std::string_view child_description,
-               opts_builder&& child_options) {
-    return add(child_run, child_name, child_description,
-               child_options.finish());
+  inline command*
+  add(std::string_view child_name, opts_builder&& child_options) {
+    return add(child_name, child_options.finish());
+  }
+
+  /// Adds a description to a command.
+  /// @returns a pointer to this command.
+  inline command* describe(std::string_view desc) {
+    description = desc;
+    return this;
+  }
+  //
+  /// Adds a Markdown-formatted documentation to a command.
+  /// @returns a pointer to this command.
+  inline command* document(std::string_view doc) {
+    documentation = doc;
+    return this;
+  }
+
+  /// Hides the command from the help text.
+  /// @returns a pointer to this command.
+  inline command* hide() {
+    visible = false;
+    return this;
+  }
+
+  /// Sets the callback for this command.
+  /// @returns a pointer to this command.
+  inline command* run(fun run, factory& fact) {
```

Comment:
I don't think this method should exist. It would be preferable to build the factory separately to the command, like it is done in `node.cpp`.